File: 442. Find All Duplicates in an Array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var findDuplicates = function(nums) {
    nums.sort();
    var repnum = [];
    var j = 0;
    for (var i = 0; i < nums.length - 1; i++) {
        if (nums[i] == nums[i + 1]) {
            repnum[j++] = nums[i];
        }
    }
    return repnum;
};

File: 442. Find All Duplicates in an Array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def find_duplicates(nums)
    nums = nums.sort
    repnum = []
    for i in 0..nums.size-1
        if nums[i] == nums[i + 1]
            repnum.push(nums[i])
        end
    end
    return repnum
end
